compiler/src/dotty/tools/dotc/cc/CaptureRef.scala

```diff
@@ -99,25 +100,56 @@ trait CaptureRef extends TypeProxy, ValueType:
    *   x: x1.type /\ x1 subsumes y  ==>  x subsumes y
    */
   final def subsumes(y: CaptureRef)(using Context): Boolean =
```

Comment:
We need to get better reassurance that this does the right thing. It looks too complicated to be able to argue it's obviously correct. I tried to run some examples, where it actually seems to do the wrong thing. So we need here

 - A formal set of rules for subcapturing.
 - A rendition of these rules as a doc comment for this method. It's a shame that the old doc comment was not updated in any way.
 - More extensive tests that try to test all combinations. I saw we have neg tests but no new pos tests.